docs/usage/16-templating.md

```diff
@@ -0,0 +1,247 @@
+            </div>
+        </body>
+    </html>
+    ```
+
+=== "Mako"
+    ```html
+    <html>
+        <body>
+            <div>
+                <span>My state value: ${request.app.state.some_key}</span>
+            </div>
+        </body>
+    </html>
+    ```
+
+### Adding CSRF inputs
+
+If you want to add a hidden `<input>` tag containing a
+[CSRF token](https://developer.mozilla.org/en-US/docs/Web/Security/Types_of_attacks#cross-site_request_forgery_csrf),
+you first need to configure [CSRF protection](/starlite/usage/7-middleware/3-builtin-middlewares/3-csrf-middleware.md).
+With that in place, you can now insert the CSRF input field inside an HTML form:
+
+=== "Jinja"
+    ```html
+    <html>
+        <body>
+            <div>
+                <form action="https://myserverurl.com/some-endpoint" method="post">
+                    {{ csrf_input }}
+                    <label for="fname">First name:</label><br>
+                    <input type="text" id="fname" name="fname">
+                    <label for="lname">Last name:</label><br>
+                    <input type="text" id="lname" name="lname">
+                </form>
+            </div>
+        </body>
+    </html>
+    ```
+
+=== "Mako"
+    ```html
+    <html>
+        <body>
+            <div>
+                <form action="https://myserverurl.com/some-endpoint" method="post">
+                    ${csrf_input}
+                    <label for="fname">First name:</label><br>
+                    <input type="text" id="fname" name="fname">
+                    <label for="lname">Last name:</label><br>
+                    <input type="text" id="lname" name="lname">
+                </form>
+            </div>
+        </body>
+    </html>
+    ```
+
+The input is hidden so users cannot see and interact with it. It will though be sent back to the server when submitted,
+and the CSRF middleware will check that the token is valid.
+
+### Passing template context
+
+Passing context to the template is very simple - its one of the kwargs expected by the [`Template`][starlite.response.TemplateResponse]
+container, so simply pass a string keyed dictionary of values:
+
+```python
+from starlite import Template, get
+
+
+@get(path="/info")
+def info() -> Template:
+    return Template(name="info.html", context={"numbers": "1234567890"})
+```
+
+
+## Template callables
+
+Both [Jinja2](https://jinja.palletsprojects.com/en/3.0.x/) and [Mako](https://www.makotemplates.org/) allow users to define custom
+callables that are ran inside the template. Starlite builds on this and offers some functions out of the box.
+
+### Built-in callables
+
+`url_for`:   To access urls for route handlers you can use the `url_for` function. Its signature and behaviour
+    matches [`route_reverse`][starlite.app.Starlite.route_reverse] behaviour. More details about route handler indexing
+    can be found [here](/starlite/usage/2-route-handlers/4-route-handler-indexing.md)
```

Comment:
The relative links seem to need the `.md` suffix, while absolute ones don't, e.g., if I change this to `[here](./2-route-handlers/4-route-handler-indexing.md)` it works, but with the absolute path seems we need to drop the `.md`.

Reply:
Ah! Finally! I couldn't make sense of it before. Admittedly, I also didn't look into it and just went with whatever seemed to work. I usually try to use absolute links now when I add new ones, since that leads to fewer broken links when stuff is moved around.